Makefile

```diff
@@ -222,3 +235,26 @@ swagger-docs: ## preview the API documentation
 		-e 'REDOC_OPTIONS=hide-hostname="true" lazy-rendering' \
 		-p $(SWAGGER_DOCS_PORT):80 \
 		bfirsh/redoc:1.6.2
+
+.PHONY: buildx
+ifeq ($(BUILDX), bundles/buildx)
+buildx: bundles/buildx # build buildx cli tool
+else
+buildx:
+endif
+
+bundles/buildx: GOOS ?= $(shell go env GOHOSTOS || uname | awk '{print tolower($$0)}' || echo windows)
+bundles/buildx: GOARCH ?= $(shell go env GOHOSTARCH || echo amd64)
+bundles/buildx: DOCKER_BUILDX_VERSION ?= v0.3.0
+bundles/buildx: DOCKER_BUILDX_REPO ?= https://github.com/docker/buildx.git
+bundles/buildx: BUILDX_DOCKERFILE ?= Dockerfile.buildx
+bundles/buildx: bundles ## build buildx CLI tool
+	# This intetionally is not using the `--output` flag from the docker CLI which is a buildkit option
+	# The idea here being that if buildx is being used, it's because buildkit is not supported natively
+	docker build -f $(BUILDX_DOCKERFILE) -t "moby-buildx:$(DOCKER_BUILDX_VERSION)" \
+		--build-arg BUILDX_COMMIT="$(DOCKER_BUILDX_VERSION)" \
+		--build-arg BUILDX_REPO="$(DOCKER_BUILDX_REPO)" \
+		--build-arg GOOS=$(GOOS) \
+		--build-arg GOARCH=$(GOARCH) \
+		. && \
+		id=$$(docker create moby-buildx:$(DOCKER_BUILDX_VERSION)) && docker cp $${id}:/usr/bin/buildx $@ && touch $@; docker rm $${id}
```

Comment:
Should we use `docker build --out=...` here?

Reply:
See the comment above this bit.
Not using `--out` because it's presuming that the person calling this doesn't have a buildkit enabled daemon.